pyomo/common/unittest.py

```diff
@@ -547,3 +554,404 @@ def assertRaisesRegex(self, expected_exception, expected_regex, *args, **kwargs)
+                    if not self.solver_available[i]:
+                        _missing.append(i)
+                return "Solver%s %s %s not available" % (
+                    's' if len(_missing) > 1 else '',
+                    ", ".join(_missing),
+                    'are' if len(_missing) > 1 else 'is',
+                )
+
+        if name in self.package_dependencies:
+            packages_ = self.package_dependencies[name]
+            if not all([self.package_available[i] for i in packages_]):
+                # Skip the test because a package is not available
+                _missing = []
+                for i in packages_:
+                    if not self.package_available[i]:
+                        _missing.append(i)
+                return "Package%s %s %s not available" % (
+                    's' if len(_missing) > 1 else '',
+                    ", ".join(_missing),
+                    'are' if len(_missing) > 1 else 'is',
+                )
+
+            # This is a hack, xlrd dropped support for .xlsx files in 2.0.1 which
+            # causes problems with older versions of Pandas<=1.1.5 so skipping
+            # tests requiring both these packages when incompatible versions are found
+            if (
+                'pandas' in self.package_dependencies[name]
+                and 'xlrd' in self.package_dependencies[name]
+            ):
+                if check_min_version(
+                    self.package_modules['xlrd'], '2.0.1'
+                ) and not check_min_version(self.package_modules['pandas'], '1.1.6'):
+                    return "Incompatible versions of xlrd and pandas"
+
+        return False
+
+    def filter_fcn(self, line):
+        """
+        Ignore certain text when comparing output with baseline
+        """
+        for field in (
+            '[',
+            'password:',
+            'http:',
+            'Job ',
+            'Importing module',
+            'Function',
+            'File',
+            'Matplotlib',
+            '-------',
+            '=======',
+            '    ^',
+        ):
+            if line.startswith(field):
+                return True
+        for field in (
+            'Total CPU',
+            'Ipopt',
+            'license',
+            'Status: optimal',
+            'Status: feasible',
+            'time:',
+            'Time:',
+            'with format cpxlp',
+            'usermodel = <module',
+            'execution time=',
+            'Solver results file:',
+            'TokenServer',
+            'function calls',
+            'List reduced',
+            '.py:',
+            '{built-in method',
+            '{method',
+            '{pyomo.core.expr.numvalue.as_numeric}',
+        ):
+            if field in line:
+                return True
+        return False
+
+    def filter_file_contents(self, lines):
+        filtered = []
+        deprecated = None
+        for line in lines:
+            # Ignore all deprecation warnings
```

Comment:
I know why we would want to ignore deprecation warnings, but do we _really_ want to?

Reply:
I'm torn.  This was added ages ago so that we didn't have to update Book baselines as frequently.  Is it better to keep updating the baselines with the deprecation warning (noting that the baseline would have to be updated *twice*: once when the deprecation was added, and again after the next release (when the version is updated)?

Reply:
My thought for why it's worth the effort is that we become starkly aware of when we are deprecating something that is mentioned in the book. It's one of those "debt vs. clear knowledge" arguments.